plugins/woocommerce/src/Internal/DataStores/Orders/DataSynchronizer.php

```diff
@@ -966,6 +975,38 @@ private function delete_auto_draft_orders() {
 		do_action( 'woocommerce_scheduled_auto_draft_delete' );
 	}
 
+	/**
+	 * Handles deletion of trashed orders after `EMPTY_TRASH_DAYS` as defined by WordPress.
+	 *
+	 * @since 8.5.0
+	 *
+	 * @return void
+	 */
+	private function delete_trashed_orders() {
+		$delete_timestamp = $this->legacy_proxy->call_function( 'time' ) - ( DAY_IN_SECONDS * EMPTY_TRASH_DAYS );
```

Comment:
I guess there's nothing technically wrong with having our own routine for deleting trashed orders regardless of datastore, but I think it'd be best if we check here that HPOS is authoritative before proceeding. We can let WP take care of legacy orders (which will propagate to HPOS when needed thanks to other code in the synchronizer).

There's `custom_orders_table_is_authoritative()` in this same class, which can be used for this purpose.

Reply:
Good point! I like returning early whenever it makes sense. Addressed in [3c37df9](https://github.com/woocommerce/woocommerce/pull/41949/commits/3c37df96b57ecd631ec62b1426776d13bd2be4a8).